### ISSUE TITLE
Minimap lighting cloak

### ISSUE THREAD

**Issue by BasedUser** [Anuken/Mindustry-Suggestions#2585]
**Describe the content or mechanics you are proposing.**

A map rule for minimaps to have a fog of war. (or light effects applied to them). Also moves player names below the lighting shader.

**Describe how you think this content will improve the game. If you're proposing new content, mention how it may add more gameplay options or how it will fill a new niche.**

This will make lighting-based PvP an option for future maps. No, really, I've seen many people make night mode maps that don't actually work because you can see everything on the minimap, including black names (using minor plugins to try and make lighting fair.)

**Before making this issue, place an `X` in the boxes below to confirm that you have acknowledged them.** *Failure to do so may result in your request being closed automatically.*

1. - [X] I have checked the [Trello](https://trello.com/b/aE2tcUwF/mindustry-trello) to make sure my suggestion isn't planned or implemented in a development version.
2. - [X] I am familiar with all the content already in the game or have glanced at the wiki to make sure my suggestion doesn't exist in the game yet.
3. - [X] I have read `README.md` to make sure my idea is not listed under the "A few things you shouldn't suggest" category.

**Comment by ShadeStrike62** [comment]
This should be added, no reason why not, it would make PvP so much better

**Comment by itcannotbe** [comment]
This should be implemented with a corresponding change in logic.

Like not allowing ulocate stuff that is in the dark.

**Comment by BasedUser** [comment]
> This should be implemented with a corresponding change in logic. Like not allowing ulocate stuff that is in the dark.

That's hard to do.

**Comment by itcannotbe** [comment]
> > This should be implemented with a corresponding change in logic. Like not allowing ulocate stuff that is in the dark.

> That's hard to do.

I know, but it's kinda stupid if you implemented FoW and allow people to use logic like FoW didn't exist.

**Comment by Humanoid-X** [comment]
logic radar hack go brrrrrr

**Comment by Ilya246** [comment]
i don't think you should do anything to logic, so logic radars become actually useful
mobile autotarget though, should not target anything in the FoW

**Comment by itcannotbe** [comment]
Logic be like: I will find anything inside the fog for you.

**Comment by LeoSko** [comment]
Why not just limit radar logics to certain radiuses within another rule along with this suggestion?

**Comment by OnyxFinger** [comment]
pretty much, a whole FoW system needs to go with it, else it wouldn't work and be a broken system, because so many things would just not care about it

**Comment by RouterChainsSuckk** [comment]
hahaha foos cleint ignoring lighting go brrrrrr

**Comment by Somka000** [comment]
> hahaha foos cleint ignoring lighting go brrrrrr

ban time

Fog of war is cool <3

**Comment by ShadeStrike62** [comment]
> > This should be implemented with a corresponding change in logic. Like not allowing ulocate stuff that is in the dark.
> 
> That's hard to do.

It really isn't though. I don't claim to be a good coder, but doing something like this could be sufficient:
-For each block, every second or so, check if it is lit up (you could average an amount of light required to increase performance).
-If it is lit up, give it the tags the let you find it with logic
-If not, logic can't find it.

I get it would take effort, but it isn't impossible...

**Comment by RouterChainsSuckk** [comment]
> > hahaha foos cleint ignoring lighting go brrrrrr
> 
> ban time
> 
> Fog of war is cool <3

haha not being obvius about it go brrr

**Comment by BasedUser** [comment]
> > That's hard to do.
> 
> It really isn't though. I don't claim to be a good coder, but doing something like this could be sufficient:
> -For each block, every second or so, check if it is lit up (you could average an amount of light required to increase performance).
> -If it is lit up, give it the tags the let you find it with logic
> -If not, logic can't find it.
> 
> I get it would take effort, but it isn't impossible...

It's a shader, that's the main problem
